streams/src/main/java/org/apache/kafka/streams/Topology.java

```diff
@@ -648,11 +648,12 @@ public synchronized Topology addSink(final String name,
      * If {@code supplier} provides stores via {@link ConnectedStoreProvider#stores()}, the provided {@link StoreBuilder}s
      * will be added to the topology and connected to this processor automatically.
      *
-     * @param name the unique name of the processor node
-     * @param supplier the supplier used to obtain this node's {@link Processor} instance
-     * @param parentNames the name of one or more source or processor nodes whose output records this processor should receive
-     * and process
-     * @return itself
+     * @param name          the unique name of the processor node
+     * @param supplier      the supplier used to construct this node's {@link Processor} instance; the implementation of supplier
+     *                      should return a newly constructed {@link Processor} instance inside the scope of the lambda expression.
```

Comment:
I think it's better to add this in the main java docs above and keep the parameter description short. What about adding a similar sentence as suggested for the web-docs into the JavaDocs? (Same for the DSL operator on `KStream`)

Also, we should not talk about lambdas, as we don't know if lambdas are used or not.

Reply:
I reverted java docs on parameter and added web-docs-like docs on main java docs for this function.